classes/Search.php

```diff
@@ -137,7 +137,23 @@ class SearchCore
 
     public static function extractKeyWords($string, $id_lang, $indexation = false, $iso_code = false)
     {
-        $sanitizedString = Search::sanitize($string, $id_lang, $indexation, $iso_code, false);
+/*         start improvement
+         it allowss the '&' to be indexed for product names like J&B*/
+        if (!stripos($string, '&')) {
+            $sanitizedString = Search::sanitize($string, $id_lang, false, Context::getContext()->language->iso_code);
+        } else {
+            $preWords = explode('&', $string);
+            $sanitizedWords = [];
+            $sanitizedString = '';
+
+            for ($i = 0; $i < count($preWords); $i++) {
+                $wd = Search::sanitize($preWords[$i], $id_lang, false, Context::getContext()->language->iso_code);
+                array_push($sanitizedWords, $wd);
+                $sanitizedString .= $i < (count($preWords) - 1) ? $preWords[$i].'&' : $preWords[$i];
```

Comment:
So if I understand it correctly, what you do is
- explode string to get each parts separated by &
- sanitize each part
- put back togethers parts, insert & between each

So in the end you sanitized everything BUT preserved the ampersands.

I think maybe rather than "not sanitize the ampersands" we could improve the function `sanitize()` to not process them.

If I am not wrong, the ampersands are removed by `html_entity_decode`, right? Line 186.

Is this `html_entity_decode` useful @PierreRambaud  ? If it's used to avoid SQL injections, I think it's a bad security safeguard 🤔  .